components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.ts

```diff
@@ -305,4 +328,68 @@ export class PolicyAddMembersComponent implements OnInit, OnDestroy {
     this.addOrRemoveQueuedMember(true, member);
     this.closeModal();
   }
+
+  public showInputs(inputName: string): boolean {
+    const typeValue = this.expressionForm.get('type').value;
+    const identityValue = this.expressionForm.get('identity').value;
+
+    this.setFormLabels(typeValue);
+
+    switch (inputName) {
+      case 'identity':
+        if (typeValue === 'user' || typeValue === 'team') {
+            this.updateValidations(inputName, true);
+            return true;
+          } else {
+            this.updateValidations(inputName, false);
+            this.expressionForm.get(inputName).reset();
+            return false;
+          }
+        break;
+      case 'name':
+        if ( typeValue === 'token' || (identityValue && identityValue !== '*') ) {
+          this.updateValidations(inputName, true);
+          return true;
+        } else {
+          this.updateValidations(inputName, false);
+          this.expressionForm.get(inputName).reset();
+          return false;
+        }
+        break;
+      default:
+        return false;
+    }
+  }
+
+  private setFormLabels(typeValue: string): void {
+    if (typeValue === 'token') {
+      this.nameOrId = 'ID';
+    } else {
+      this.allIdentity = typeValue;
+      this.nameOrId = 'Name';
+    }
+  }
+
+  private updateValidations(inputName: string, active: boolean): void {
+    const updateInput = this.expressionForm.get(inputName);
+
+    if (active) {
+      return inputName === 'name'
```

Comment:
This does not quite make sense; why are you trying to return something from a function that declares it returns void? Also, by returning early, you are not executing L387 (which the docs say must follow a call to `setValidators`).
I think you just want this to be an `if (input === 'name')`